src/main/java/jarvis/storage/JsonAdaptedLesson.java

```diff
@@ -0,0 +1,242 @@
+            jsonAdaptedStudentList.add(new JsonAdaptedStudent(student));
+        }
+        this.studentList = jsonAdaptedStudentList;
+        this.attendance = attendance;
+        this.generalNotes = generalNotes;
+        this.studentNotes = studentNotes;
+        this.isCompleted = isCompleted;
+    }
+
+    /**
+     * Constructs a {@code JsonAdaptedConsult}, {@code JsonAdaptedMasteryCheck} or
+     * {@code JsonAdaptedStudio} with the given lesson details.
+     *
+     * @param lesson The given lesson, which could be a {@code Consult}, {@code MasteryCheck} or {@code Studio}.
+     * @return The Jackson-friendly version of the lesson.
+     */
+    public static JsonAdaptedLesson createLesson(Lesson lesson) {
+        CollectionUtil.requireAllNonNull(lesson);
+        assert (lesson instanceof Consult || lesson instanceof MasteryCheck) || lesson instanceof Studio;
+        if (lesson instanceof Consult) {
+            Consult consult = (Consult) lesson;
+            return new JsonAdaptedConsult(consult);
+        } else if (lesson instanceof MasteryCheck) {
+            MasteryCheck masteryCheck = (MasteryCheck) lesson;
+            return new JsonAdaptedMasteryCheck(masteryCheck);
+        } else if (lesson instanceof Studio) {
+            Studio studio = (Studio) lesson;
+            return new JsonAdaptedStudio(studio);
+        } else {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    protected String getLessonDesc() {
+        return lessonDesc;
+    }
+
+    protected LocalDateTime getStartDateTime() {
+        return startDateTime;
+    }
+
+    protected LocalDateTime getEndDateTime() {
+        return endDateTime;
+    }
+
+    protected List<JsonAdaptedStudent> getStudentList() {
+        return studentList;
+    }
+
+    protected Map<Integer, Boolean> getAttendance() {
+        return attendance;
+    }
+
+    protected ArrayList<String> getGeneralNotes() {
+        return generalNotes;
+    }
+
+    protected Map<Integer, ArrayList<String>> getStudentNotes() {
+        return studentNotes;
+    }
+
+    protected boolean isCompleted() {
+        return isCompleted;
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted lesson object into the model's {@code Lesson} object.
+     *
+     * @throws IllegalArgumentException if there were any data constraints violated in the adapted lesson.
+     */
+    public abstract Lesson toModelType() throws IllegalArgumentException;
+
+
+    /**
+     * Checks whether the argument provided is null.
+     *
+     * @param className The class whose constructor is intended to take in the given argument.
+     * @param missingFieldFormat The missing field message used in the error message.
+     * @param arg The argument to check.
+     * @throws IllegalArgumentException If the argument provided is null.
+     */
+    public static void checkNullArgument(Class<?> className, String missingFieldFormat, Object arg)
```

Comment:
I think maybe this can be shifted to util? I noticed u imported this function in `JsonAdaptedStudent` and it seems like there is nothing related to `Lesson` in this function.

Reply:
Also is there a reason why you chose to use this instead of `requireNonNull`?

Reply:
> I think maybe this can be shifted to util? I noticed u imported this function in `JsonAdaptedStudent` and `JsonAdaptedTask` and it seems like there is nothing related to `Lesson` in this function.

Initially I felt that this method didn't really fit in with the other methods in JsonUtil, so I left it in JsonAdaptedLesson first. I do agree that it shouldn't be in the JsonAdaptedLesson class though, I will change it accordingly

Reply:
> Also is there a reason why you chose to use this instead of `requireNonNull`?

requireNonNull() throws a NullPointerException and in this case an IllegalArgumentException should be thrown instead, with a custom error message. Hence I think it is more efficient to use this function especially since the requireNonNull() function does not provide a way to have a custom error message.